src/main/java/bookface/model/book/Book.java

```diff
@@ -94,12 +97,25 @@ public void loanTo(Person loanee, Date returnDate) {
 
 
     /**
-     * Return this loaned book .
+     * Return this loaned book.
      */
     public void markBookAsReturned() {
         this.loanee = null;
     }
 
+    /**
+     * Checks if the book is overdue.
+     * @return True if book is overdue, false otherwise
+     */
+    public boolean isOverdue() {
+        if (this.isLoaned()) {
+            long millis = System.currentTimeMillis();
+            Date todayDate = new Date(millis);
+            return todayDate.after(this.getReturnDate());
+        }
+        return false;
```

Comment:
Hmm actually thinking about it, what does it mean to return false if the book is not loaned? Should we retuan an Optional<Boolean>? So if the book is not loaned, we return Optional.empty() instead of returning false, which can be mistakenly interpreted as a book not being overdue, even when it is not on loan

Reply:
In this case I think there's no difference since all method calls check for the loan status before checking for overdue status, but yes, your implementation is better

Reply:
True, I was thinking if we make any changes in the future and we forget it, handling the Optional might prevent bugs